feat:Adding ability to send back picture of Id document (#90)
fiatconnect/specification: fiatconnect-api.md

  - [9.3. Initial Entity Support](#93-initial-entity-support)
    - [9.3.1. KYC Schemas](#931-kyc-schemas)
      - [9.3.1.1. `PersonalDataAndDocuments`](#9311-personaldataanddocuments)
      - [9.3.1.2. `PersonalDataAndDocumentsDetailed`](#9312-personaldataanddocumentsdetailed)

>         - [9.3.1.2.1. `]

Reviewer: ```suggestion
        - [9.3.1.2.1. `IdentificationDocumentType`](#93121-identificationdocumenttype)
```

Author: Corrected, again my oversight

	},
	phoneNumber: `string`,
	selfieDocument: `string`,

>     identificationDocument: `string`

Reviewer: nit: indentation

}
```

The `selfieDocument` and `identificationDocument` fields should be base64 encoded binary blobs representing images.

#### 9.3.1.2. `PersonalDataAndDocumentsDetailed`

> A more detailed Schema allowing the user to provide more information to give them a greater chance to be admitted.

Reviewer 2: ```suggestion
A more detailed Schema allowing the user to provide more information to work with CICO providers that require it.
```
Seems like "to give them a greater chance to be admitted" is misleading-- sounds like the user can optionally provide more info, whereas some providers may *only* accept KYC attempts with this extra set of information

```
{
	firstName: `string`,
	middleName?: `string`,
	lastName: `string`,
	dateOfBirth: {
		day: `string`,
		month: `string`,
		year: `string`
	},
	address: {
		address1: `string`,
		address2?: `string`,
		isoCountryCode: `string`,
		isoRegionCode: `string`,
		city: `string`,
		postalCode?: `string`
	},
	phoneNumber: `string`,

>     email: `string`,

Reviewer: nit: indentation in this schema

>     email?: `string`,

Reviewer 2: ```suggestion
    email: `string`,
```
per @ailsoncgt 's request in [Discord](https://discord.com/channels/1020413508350906470/1020413509172994160/1066046333384937512)

	selfieDocument: `string`,

>     identificationDocumentType: `enum`,

Reviewer: ```suggestion
    identificationDocumentType: `IdentificationDocumentTypeEnum`,
```

	identificationDocumentFront: `string`,
    identificationDocumentBack?: `string` 
}
```

The `phoneNumber` field is REQUIRED and MUST follow the formating of the [E.164 international standard](https://en.wikipedia.org/wiki/E.164). It MUST contain all parts of the phone number INCLUDING the Country Code.

> The `email` field is OPTIONAL but if provided MUST be a [valid email](https://en.wikipedia.org/wiki/Email_address#Syntax).

Reviewer 2: ```suggestion
The `email` field MUST be a [valid email](https://en.wikipedia.org/wiki/Email_address#Syntax).
```

> The accepted documents can be found [here](https://en.wikipedia.org/wiki/Identity_document). The document should fit one of the `identificationDocumentType` field options:

Reviewer: Can we formally add an `IdentificationDocumentType` enum to the spec? It would be consistent with patterns used elsewhere in the spec, and we would presumably create an `IdentificationDocumentType` enum in the `fiatconnect-types` repo anyhow, so it would be good to have it mirrored in the specifcation.

> The accepted documents can be found [here](https://en.wikipedia.org/wiki/Identity_document). The document should fit one of the `identificationDocumentType` field options:

Reviewer: ```suggestion
The accepted documents can be found [here](https://en.wikipedia.org/wiki/Identity_document). The document MUST fit one of the `identificationDocumentType` field options:
```

- `IDC`: State issued identity card
- `PAS`: Passport

> - `DL`: Driving Liscense

Reviewer: ```suggestion
- `DL`: Driving License
```

> The `selfieDocument`, `identificationDocumentFront` and `identificationDocumentFront` fields should be base64 encoded binary blobs representing images.

Reviewer: ```suggestion
The `selfieDocument`, `identificationDocumentFront` and `identificationDocumentFront` fields MUST be base64 encoded binary blobs representing images.
```

> The `identificationDocumentBack` field is REQUIRED if the `identificationDocumentType` is `IDC` or `DL`

Reviewer 2: ```suggestion
The `identificationDocumentBack` field is REQUIRED if the `identificationDocumentType` is `IDC` or `DL`. Otherwise, it is optional.
```

Reviewer 2: Not important, just thought this could be a little clearer.

> 

Reviewer: You can add this section here to specify the enum I mentioned in another comment.

`````suggestion
##### 9.3.1.2.1 `IdentificationDocumentType`

`IdentificationDocumentType` is used to represent the type of document being submitted with a `PersonalDataAndDocumentsDetailed` KYC schema.

```
[
  `IDC`,
  `PAS`,
  `DL`
]
```

The enum values represent the following forms of identification:
* `IDC`: State-issued identity card
* `PAS`: Passport
* `DL`: Driver's License
`````

	},
	phoneNumber: `string`,
	selfieDocument: `string`,

>   identificationDocument: `string`

Reviewer: nit: indentation (I don't think I can add tab characters to suggestions)

Author: Should be fixed, the file is not consistent as a whole though, sometime there is use of tab, sometime there is double space: 
![image](https://user-images.githubusercontent.com/6547517/215056788-1f918ee9-4e2d-41a0-aaae-90e6ff6e39d8.png)

and 

![image](https://user-images.githubusercontent.com/6547517/215056951-73e15853-69fd-47c5-930b-e1fd8965fb46.png)

}
```

The `selfieDocument` and `identificationDocument` fields should be base64 encoded binary blobs representing images.

#### 9.3.1.2. `PersonalDataAndDocumentsDetailed`
A more detailed Schema allowing the user to provide more information to work with CICO providers that require it.

```
{
	firstName: `string`,
	middleName?: `string`,
	lastName: `string`,
	dateOfBirth: {
		day: `string`,
		month: `string`,
		year: `string`
	},
	address: {
		address1: `string`,
		address2?: `string`,
		isoCountryCode: `string`,
		isoRegionCode: `string`,
		city: `string`,
		postalCode?: `string`
	},
	phoneNumber: `string`,

>   email?: `string`,

Reviewer: indentation here and elsewhere in this schema

	selfieDocument: `string`,
  identificationDocumentType: `IdentificationDocumentTypeEnum`,
	identificationDocumentFront: `string`,
  identificationDocumentBack?: `string` 
}
```

> The `phoneNumber` field is REQUIRED and MUST follow the formating of the [E.164 international standard](https://en.wikipedia.org/wiki/E.164). It MUST contain all parts of the phone number INCLUDING the Country Code.

Reviewer: `INCLUDING` is not a keyword as specified by BCP 14.
```suggestion
The `phoneNumber` field is REQUIRED and MUST follow the formating of the [E.164 international standard](https://en.wikipedia.org/wiki/E.164). It MUST contain all parts of the phone number including the Country Code.
```

Author: Corrected

The `email` field MUST be a [valid email](https://en.wikipedia.org/wiki/Email_address#Syntax).

> The `identificationDocumentType` field must be selected from `IdentificationDocumentTypeEnum`, and is used to represent the kind of document being submitted.

Reviewer: ```suggestion
The `identificationDocumentType` field MUST be selected from `IdentificationDocumentTypeEnum`, and is used to represent the kind of document being submitted.
```

Author: Corrected

> The `selfieDocument`, `identificationDocumentFront` and `identificationDocumentFront` fields MUST be base64 encoded binary blobs representing images.

Reviewer: ```suggestion
The `selfieDocument`, `identificationDocumentFront` and `identificationDocumentBack` fields MUST be base64 encoded binary blobs representing images.
```

Author: Corrected

> The `identificationDocumentBack` field is REQUIRED if the `identificationDocumentType` is `IDC` or `DL`. Otherwise, it is optional.

Reviewer: `OPTIONAL` is a BCP 14 keyword.
```suggestion
The `identificationDocumentBack` field is REQUIRED if the `identificationDocumentType` is `IDC` or `DL`. Otherwise, it is OPTIONAL.
```

Author: Corrected
